components/TextAreaDetails.tsx

```diff
@@ -9,35 +9,58 @@ import { Details } from "./Details";
 import { TextArea } from "./TextArea";
 
 type Props = DynamicComponentControlledProps<string> & {
-  summary: string;
+  summary: React.ReactNode;
   name: string;
   rows?: number | null;
+  label?: {
+    id?: string | null;
+    value?: React.ReactNode | null;
+  };
+  contentAfterTextArea?: React.ReactNode;
 };
 
 export const TextAreaDetails: React.FunctionComponent<Props> = (
   props
 ): React.ReactElement => {
-  const { summary, name, rows, value, onValueChange, disabled } = props;
+  const {
+    summary,
+    name,
+    rows,
+    label,
+    contentAfterTextArea,
+    value,
+    onValueChange,
+    disabled
+  } = props;
 
   const summaryId = `${name}-summary`;
 
   return (
     <Details summary={{ id: summaryId, value: summary }}>
       <TextArea
-        label={{ id: summaryId }}
+        label={{
+          id: label && label.id ? label.id : summaryId,
+          value: label && label.value ? label.value : null
```

Comment:
We generally prefer `undefined` over `null` where possible. I think this can be simplified to:

```suggestion
          value: label && label.value
```